index.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react'
+
+import { findNodeHandle, NativeModules, Platform, TextInput, TextInputProps } from 'react-native'
+const { RNTextInputMask } = NativeModules as { RNTextInputMask: MaskOperations }
+export const { mask, unmask, setMask } = RNTextInputMask
+
+const TextInputMask = forwardRef<Handles, TextInputMaskProps>(({ maskDefaultValue = true, mask: inputMask, value: defaultValue, multiline, onChangeText, ...rest }, ref) => {
+  const input = useRef<TextInput>(null)
+  const [value, setValue] = useState<string>()
+
+  useEffect(() => {
+    if (maskDefaultValue && inputMask && defaultValue) {
+      mask(inputMask, `${defaultValue}`, (text) =>
+          setValue(text)
+      )
+    } else if (!inputMask) {
+      setValue(defaultValue)
+    }
+    const nodeId = findNodeHandle(input.current)
+    if (inputMask && nodeId) {
+      setMask(nodeId, inputMask)
+    }
+  }, [maskDefaultValue, inputMask, defaultValue])
+
+  useImperativeHandle(ref, () => ({
+    focus: () => {
```

Comment:
exposing focus handlers, something we use and i figured it doesn't hurt and might be useful for others